pkg/core/server/connect.go

```diff
@@ -880,48 +872,57 @@ func (c *CoreService) GetRewardAttestation(ctx context.Context, req *connect.Req
 	}
 
 	// Get programmatic reward by deployed address
-	reward, err := c.core.db.GetReward(ctx, rewardAddress)
+	dbReward, err := c.core.db.GetReward(ctx, rewardAddress)
 	if err != nil {
 		return nil, connect.NewError(connect.CodeNotFound, errors.New("programmatic reward not found"))
 	}
 
-	sigData := &v1.RewardAttestationSignature{
-		EthRecipientAddress: req.Msg.EthRecipientAddress,
-		Amount:              req.Msg.Amount,
-		RewardAddress:       req.Msg.RewardAddress,
-		RewardId:            req.Msg.RewardId,
-		Specifier:           req.Msg.Specifier,
-		ClaimAuthority:      req.Msg.ClaimAuthority,
+	// Convert DB reward to rewards package format
+	claimAuthorities := make([]rewards.ClaimAuthority, len(dbReward.ClaimAuthorities))
+	for i, ca := range dbReward.ClaimAuthorities {
+		claimAuthorities[i] = rewards.ClaimAuthority{
+			Address: ca,
+			Name:    "", // Name not stored in DB
+		}
 	}
 
-	signer, err := common.ProtoRecover(sigData, req.Msg.Signature)
-	if err != nil {
-		return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("signature recovery failure"))
+	reward := rewards.Reward{
+		ClaimAuthorities: claimAuthorities,
+		Amount:           uint64(dbReward.Amount),
+		RewardId:         dbReward.RewardID,
+		Name:             dbReward.Name,
 	}
 
-	// Verify claim authority is authorized to sign for this reward
-	authorized := false
-	for _, ca := range reward.ClaimAuthorities {
-		if strings.EqualFold(ca, signer) {
-			authorized = true
-			break
-		}
+	// Create claim for validation (without RewardAddress to maintain backward compatibility)
+	claim := rewards.RewardClaim{
+		RecipientEthAddress:       ethRecipientAddress,
+		Amount:                    amount,
+		RewardID:                  req.Msg.RewardId,
+		Specifier:                 specifier,
+		AntiAbuseOracleEthAddress: claimAuthority, // Using claimAuthority as oracle for programmatic rewards
```

Comment:
could rename this param on RewardClaim to just be `claimAuthority` going forward, and say that the AAO is the claim authority for the old rewards 🤷